src/utils.js

```diff
@@ -35,3 +37,16 @@ const getBrowserName = (): BrowserNames => parser.getResult().browser.name;
 const getMajor = (): number => parseInt(parser.getResult().browser.major, 10);
 
 export const isIE9 = () => getBrowserName() === 'IE' && getMajor() <= 9;
+
+export const burnItDown = (message?: BurnItDownResponse) => {
+  let timeToBurnItDown =
+    message && !message.force && message.before ? message.before - new Date().getTime() : 0;
+  if (timeToBurnItDown < 0) {
+    timeToBurnItDown = 0;
+  }
+
+  setTimeout(() => {
+    unsubscribe();
+    setBurned();
```

Comment:
Nitpick-ish: should `setBurned` be called first, so there's absolutely nothing getting in the way of blocking future requests? I'm not entirely sure how complex Atmosphere's disconnection code is or if it's possible for it to get in a state where it would fail.